### DISCUSSION TITLE
Can’t run Rust tests?

### DISCUSSION THREAD

**Discussion by codahale** [zed-industries/zed#22523] [Feature Requests]
Hi all,

Just started trying out Zed (0.166.1) on macOS 15.2 and there’s a lot to like, but when I run tests in a Rust project (i.e. push the little play button annotation next to the test function or module), `cargo` spits out a bunch of errors like this:

```
error: could not compile `anyhow` (lib) due to 1 previous error
error: couldn't create a temp dir: No such file or directory (os error 2) at path "/var/folders/s1/rv5w7wln4jq71wgx1mgt0g2r0000gn/T/\n\u{1b}[?2004l/rustcm9cnZK"
```

Running `cargo test -p blah bloo -- --nocapture` in the terminal works exactly as I’d expect.

The directory `/var/folders/s1/rv5w7wln4jq71wgx1mgt0g2r0000gn/T` exists, and it has a number of subdirectories which seem based on the same pattern as `rustcm9cnZK`, but `\n\u{1b}[?2004l` seems like a deeply weird directory name.

Any ideas? Should I open an issue?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
